src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -120,14 +120,10 @@ export function DiscoverMainRoute({
         const { dataSource } = stateContainer.appState.getState();
         const isEsqlQuery = isDataSourceType(dataSource, DataSourceType.Esql);
 
-        // ES|QL should work without data views
-        // Given we have a saved search id, we can skip the data/data view check, too
-        // A given nextDataView is provided by the user, and therefore we can skip the data/data view check
-
         if (savedSearchId || isEsqlQuery || nextDataView) {
-          if (!isEsqlQuery) {
-            await stateContainer.actions.loadDataViewList();
-          }
+          // even ES|QL doesn't need a data view, we still need to load the data view list to
```

Comment:
```suggestion
          // Although ES|QL doesn't need a data view, we still need to load the data view list to
```

If ES|QL mode does not need other data views then we could optimize it in the future by loading data views only after the switch to Data view mode. This would save some initial loading time for Discover in ES|QL mode.

Reply:
yes, agred, this is how it should work, and should be a done this way in the future 👍